src/energyplus/ForwardTranslator/ForwardTranslateEnergyManagementSystemActuator.cpp

```diff
@@ -65,113 +69,203 @@ namespace openstudio {
 namespace energyplus {
 
   boost::optional<IdfObject> ForwardTranslator::translateEnergyManagementSystemActuator(EnergyManagementSystemActuator& modelObject) {
-    boost::optional<std::string> s;
 
-    IdfObject idfObject(openstudio::IddObjectType::EnergyManagementSystem_Actuator);
+    const boost::optional<ModelObject> actuatedComponent_ = modelObject.actuatedComponent();
+    if (!actuatedComponent_) {
+      LOG(Error, "Actuated Component for Actuator " << modelObject.nameString() << " does not exist, it will not be translated.");
+      return boost::none;
+    }
 
-    //Name
-    s = modelObject.name();
-    if (s) {
-      idfObject.setName(*s);
+    // Start by dealing with "OS:Site" and "OS:WeatherFile" that are special ones (will not pass the test on `m.name()` below
+    if (actuatedComponent_->iddObjectType() == openstudio::IddObjectType::OS_Site
+        || actuatedComponent_->iddObjectType() == openstudio::IddObjectType::OS_WeatherFile) {
+      IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::EnergyManagementSystem_Actuator, modelObject);
+      idfObject.setString(EnergyManagementSystem_ActuatorFields::ActuatedComponentUniqueName, "Environment");
+      idfObject.setString(EnergyManagementSystem_ActuatorFields::ActuatedComponentType, "Weather Data");
+      idfObject.setString(EnergyManagementSystem_ActuatorFields::ActuatedComponentControlType, modelObject.actuatedComponentControlType());
+      return idfObject;
```

Comment:
Throughout the file, I'm going to be heavily using early returns: either a boost::none when we can't recover, or a valid IdfObject. I wanted to avoid nesting too much and ending up with 32 spaces of indent